trie/trie.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include <cstring>
+using namespace std;
+
+// Determine size of array
+#define ARRAY_SIZE(a) sizeof(a)/sizeof(a[0])
+// Alphabet size (# of symbols)
+#define ALPHABET_SIZE 26
+// Convert char to int value
+// only 'a' to 'z' are allowed chars
+#define CHAR_TO_INT(c) ((int)c - (int)'a')
+
+// TrieNode class
+class TrieNode {
+	private:
+		// Children of every Node
+		class TrieNode *children[ALPHABET_SIZE];
+		// Status of Leaf Node
+		bool isLeaf;
+	public:
+
+		// Default Constructor
+		TrieNode() {
+			this->isLeaf = false;
+			for (int i = 0; i < ALPHABET_SIZE; ++i)
+				this->children[i] = NULL;
+		}
+
+		// Get leaf node status
+		bool get_leaf_status() { return isLeaf; }
+		// Set leaf node status
+		void set_leaf_status(bool status) { this->isLeaf = status; }
+		// Get Child status
+		// Check if it is null or not
+		bool get_child_status(int index) {
+			if (!(this->children[index]))
+				return true;
+			return false;
+		}
+
+		// Initialize the child node, at specific index
+		void initialize_child(int index) {
+			this->children[index] = new TrieNode();
+		}
+
+		// Get child node at specific index
+		class TrieNode *get_children(int index) {
+			return this->children[index];
+		}
+
+		// Insert into the Trie, Implemetation is below
+		void insert(string key);
+
+		// Search into the Trie, Implementation is below
+		bool search(string key);
+};
+
+// Insert into the Trie
+// args :
+//    key : String to insert into Trie
+// Time Complexity: O(len(key))
```

Comment:
Use `/* ... */` blocks for function comments. See https://stackoverflow.com/questions/51667/best-tips-for-documenting-code-using-doxygen